osbuild/remoteloop.py

```diff
@@ -112,9 +112,10 @@ class LoopClient:
     def __init__(self, sock):
         self.sock = sock
 
-    def create_device(self, fd, offset=None, sizelimit=None):
+    def create_device(self, filename, offset=None, sizelimit=None):
```

Comment:
Hm, this makes the library less generally useful (I'd like to ideally allow people to reuse this library), and in general paths are nasty beasts. And unless it makes other patches simpler there does not appear to be a clear benefit?

Reply:
The benefit is mostly convenience: We'd like to open the file with `os.open` so that we can pass `O_DIRECT`, but `os.open` isn't a context manager. I had some weird-looking code in `qemu` and `rawfs` to make this work. Moving it into this function seemed more clean to me.


I see your point, though. What do you think about allowing both a filename or an fd, similar to [open](https://docs.python.org/3/library/functions.html#open)? Or would you like to encourage not passing paths at all?

Reply:
Ah, got it. In that case let's keep it this way. No need to bend over backwards to make this work with a context object just for the sake of it.

Reply:
Thanks! Makes sense, let's add it when we need it. I do like the "allowing both" approach and it seem to be something the standard library has been moving to.